### PR TITLE
Refactor: PeginQuote class definition

### PR DESCRIPTION
**Pull request by ronaldsg20**
I've defined the PeginQUote class in order to assing all the required
		behavior to each specific quote object

### CODE REVIEW

File: src/pegin/components/create/ConfirmationSteps.vue [hunk 1]
```diff
@@ -434,18 +434,12 @@ export default defineComponent({
...
 
-    function getProviderFee(): SatoshiBig {
-      return selectedQuote.value.quote.productFeeAmount
-        .plus(selectedQuote.value.quote.callFee)
-        .plus(SatoshiBig.fromWeiBig(selectedQuote.value.quote.gasFee));
-    }
-
     const amountToTransfer = computed(() => (flyover.value
-      ? selectedQuote.value.quote.value.plus(getProviderFee())
+      ? selectedQuote.value.quote.value.plus(selectedQuote.value.providerFee)
```
Comment (lserra-iov) [inline src/pegin/components/create/ConfirmationSteps.vue:442]: We probably want to use `totalValueToTransfer` here too.

**Review by lserra-iov — APPROVED** [review]

File: src/pegin/components/create/ConfirmTx.vue [hunk 2]
```diff
@@ -146,13 +141,13 @@ export default defineComponent({
...
 
-    const flyoverTotalFee = computed(() => getProviderFee()
+    const flyoverTotalFee = computed(() => selectedQuote.value.providerFee
       .plus(safeFee.value));
 
     const flyoverPeginSummary = computed((): NormalizedSummary => ({
       amountFromString: selectedQuote.value.quote.value.toBTCTrimmedString(),
       amountReceivedString: selectedQuote.value.quote.value.toBTCTrimmedString(),
-      fee: Number(flyoverTotalFee.value),
+      fee: Number(flyoverTotalFee.value.toBTCTrimmedString()),
```
Comment (alexjavabraz) [inline src/pegin/components/create/ConfirmTx.vue:150]: flyoverTotalFee needs to be on selectedQuote object.

File: src/pegin/components/create/ConfirmationSteps.vue [hunk 3]
```diff
@@ -434,18 +434,12 @@ export default defineComponent({
...
     const flyover = computed(() => pegInTxState.value.peginType === constants.peginType.FLYOVER);
     const changeIdx = flyover.value ? 1 : 2;
 
-    function getProviderFee(): SatoshiBig {
-      return selectedQuote.value.quote.productFeeAmount
-        .plus(selectedQuote.value.quote.callFee)
-        .plus(SatoshiBig.fromWeiBig(selectedQuote.value.quote.gasFee));
-    }
-
     const amountToTransfer = computed(() => (flyover.value
```
Comment (alexjavabraz) [inline src/pegin/components/create/ConfirmationSteps.vue:441]: that calculation needs to be internally on PeginQuote object.

Comment (alexjavabraz) [inline src/pegin/components/create/ConfirmationSteps.vue:441]: this calculation needs to be internally on the PeginQuote object

File: src/pegin/components/create/PegInForm.vue [hunk 4]
```diff
@@ -144,16 +144,11 @@ export default defineComponent({
     const selectedAccountBalance = useGetter<SatoshiBig>('pegInTx', constants.PEGIN_TX_GET_SELECTED_BALANCE);
 
     const enoughAmountFlyover = computed(() => {
-      const quote = selectedQuote.value?.quote;
-      if (!quote) {
+      if (!selectedQuote.value) {
```
Comment (alexjavabraz) [inline src/pegin/components/create/PegInForm.vue:147]: create a function like isValid internally on `selecterQuote` object, verifying if value is not empty or 0 or null.

File: src/pegin/components/create/PegInForm.vue [hunk 5]
```diff
@@ -144,16 +144,11 @@ export default defineComponent({
...
         return false;
       }
-      const fullAmount: SatoshiBig = quote?.value
-        .plus(quote.productFeeAmount)
-        .plus(quote.callFee)
-        .plus(SatoshiBig.fromWeiBig(quote.gasFee))
+      const fullAmount: SatoshiBig = selectedQuote.value.totalValueToTransfer
```
Comment (alexjavabraz) [inline src/pegin/components/create/PegInForm.vue:150]: insert selectedFee on `selectedQuote` object, and then call getFullAmount from `selectedQuote` object

File: src/pegin/components/create/PeginOptionCard.vue [hunk 6]
```diff
@@ -125,11 +123,11 @@ export default defineComponent({
         label: 'Powered by PowPeg + Flyover',
         subtitleColor: 'orange',
         link: 'https://dev.rootstock.io/concepts/rif-suite/#meet-the-suite',
-        estimatedTime: () => blockConfirmationsToTimeString(quote.value?.confirmations ?? 0, 'btc'),
-        amountToTransfer: () => quote.value?.value
-          .plus(quoteFee.value).plus(selectedFee.value) ?? new SatoshiBig('0', 'btc'),
+        estimatedTime: () => blockConfirmationsToTimeString(computedQuote.value?.quote.confirmations ?? 0, 'btc'),
+        amountToTransfer: () => computedQuote.value?.totalValueToTransfer
```
Comment (alexjavabraz) [inline src/pegin/components/create/PeginOptionCard.vue:127]: you can use the same function mentioned above `getFullAmount`

File: src/pegin/components/create/PeginOptionCard.vue [hunk 7]
```diff
@@ -125,11 +123,11 @@ export default defineComponent({
...
+          .plus(selectedFee.value) ?? new SatoshiBig('0', 'btc'),
         providerFee: () => quoteFee.value,
-        valueToReceive: () => quote.value?.value ?? new SatoshiBig('0', 'btc'),
+        valueToReceive: () => computedQuote.value?.quote.value ?? new SatoshiBig('0', 'btc'),
```
Comment (alexjavabraz) [inline src/pegin/components/create/PeginOptionCard.vue:130]: you can use the same function mentioned above `isValid`

File: src/pegin/components/create/ConfirmTx.vue (hunk 2 above)
Comment (ronaldsg20) [reply src/pegin/components/create/ConfirmTx.vue:150]: I'm assuming the quote is not responsable of the networkFee calculation, here we called `safeFee` and its outside of the Quote

**Review by alexjavabraz — APPROVED** [review]
LGTM

**Review by annipi — APPROVED** [review]
LGTM!

**Review by leoiovlabs — APPROVED** [review]
LGTM

**Pull request merged**